tests/aircall/test_aircall.py

```diff
@@ -1,84 +1,219 @@
 import pytest
-
+from pydantic import ValidationError
+
+from tests.aircall.helpers import build_con_and_ds, handle_mock_data
+from tests.aircall.mock_results import (
+    fake_tags,
+    fake_teams,
+    fake_users,
+    filtered_calls,
+    filtered_tags,
+    filtered_teams,
+    filtered_users,
+)
 from toucan_connectors.aircall.aircall_connector import AircallConnector, AircallDataSource
 
+columns_for_calls = [
+    'id',
+    'direction',
+    'duration',
+    'answered_at',
+    'ended_at',
+    'raw_digits',
+    'user_id',
+    'tags',
+    'user_name',
+    'team',
+    'day',
+]
+columns_for_tags = ['id', 'name', 'color', 'description']
+columns_for_teams = ['team', 'user_id', 'user_name', 'user_created_at']
+columns_for_users = ['user_id', 'user_name', 'user_created_at']
+
+fetch_fn_name = 'toucan_connectors.aircall.aircall_connector.fetch_page'
+
 
 @pytest.fixture
 def con(bearer_aircall_auth_id):
     return AircallConnector(name='test_name', bearer_auth_id=bearer_aircall_auth_id)
 
 
-@pytest.mark.flaky(reruns=5, reruns_delay=2)
-def test_aircall_params_default_limit(con, mocker):
-    """It should retrieve 100 entries by default"""
-    get_page_data_spy = mocker.spy(AircallConnector, '_get_page_data')
-    ds = AircallDataSource(
-        name='test_name', domain='test_domain', endpoint='/calls', filter='.calls | map({id})',
+@pytest.mark.asyncio
+async def test__get_data_tags_case(mocker):
+    """Tests with tags happy case"""
+    dataset = 'tags'
+    fake_res = handle_mock_data(fake_tags)
+    fake_fetch_page = mocker.patch(fetch_fn_name, return_value=fake_res)
+    con, ds = build_con_and_ds(dataset)
+    res = await con._get_tags(ds.dataset, {}, 10)
+
+    assert fake_fetch_page.call_count == 1
+    assert len(res) == 3
+
+
+@pytest.mark.asyncio
+async def test__get_data_tags_unhappy_case(mocker):
+    """Tests what happens when tags call returns an error"""
+    dataset = 'tags'
+    mocker.patch(fetch_fn_name, return_value=Exception('OMGERD OOPS!!!'))
+    con, ds = build_con_and_ds(dataset)
+    with pytest.raises(Exception):
+        await con._get_tags(ds.dataset, {}, 10)
+
+
+@pytest.mark.asyncio
+async def test__get_data_users_case(mocker):
+    """Tests users call happy case"""
+    dataset = 'users'
+    fake_res = handle_mock_data([fake_teams, fake_users])
+    fake_fetch_page = mocker.patch(fetch_fn_name, side_effect=fake_res)
+    con, ds = build_con_and_ds(dataset)
+    res = await con._get_data(ds.dataset, {}, 10)
+
+    assert fake_fetch_page.call_count == 2
+    assert len(res) == 2
+    teams, users = res
```

Comment:
Hmm, since we're mocking and using fake data, couldn't we just test that `assert teams == fake_teams`?